code/src/Reducers/labyrinth.js

```diff
@@ -0,0 +1,120 @@
+// /////////////// IMPORTS //////////////////////// //
+
+// Import necessary libraries
+import { createSlice } from '@reduxjs/toolkit';
+import { v4 as uuidv4 } from 'uuid';
+
+// Set initial state for the store
+const initialState = {
+  userName: '',
+  name: '',
+  description: '',
+  direction: '',
+  moves: '',
+  history: [],
+  loading: false,
+  coordinates: ''
+}
+
+// /////////////// REDUCER //////////////////////// //
+
+// Create Redux slice
+export const labyrinthMango = createSlice({
+  name: 'labyrinthMango',
+  initialState,
+  reducers: {
+    // Set user name and generate unique user ID
+    setUserName: (store, action) => {
+      const input = action.payload;
+      const userId = uuidv4(input);
```

Comment:
What does this mean? `const userId = uudv4(input)` We only had username but I see you have username, name and userID, how does that work? Is it a way of making the username random?